libcxx/include/__format/format_arg_store.h

```diff
@@ -101,20 +101,14 @@ consteval __arg_t __determine_arg_t() {
   return __arg_t::__long_double;
 }
 
-// Char pointer
+// Char pointer or array
 template <class _Context, class _Tp>
-  requires(same_as<typename _Context::char_type*, _Tp> || same_as<const typename _Context::char_type*, _Tp>)
+  requires(same_as<typename _Context::char_type*, _Tp> || same_as<const typename _Context::char_type*, _Tp>) ||
+          (is_array_v<_Tp> && same_as<_Tp, typename _Context::char_type[extent_v<_Tp>]>)
```

Comment:
Doesn't this form the invalid type `_Context::char_type[0]` if `_Tp` is not an array (because `extent_v<_Tp>` will then be 0)?

Reply:
Yes, although this is pre-existing. I think we should handle character arrays more properly.